packages/oui-dual-list/src/dual-list.controller.js

```diff
@@ -0,0 +1,251 @@
+import { addBooleanParameter, addDefaultParameter } from "@oui-angular/common/component-utils";
+export default class {
+    constructor ($q, $element, $attrs, ouiDualListProvider) {
+        "ngInject";
+
+        this.$q = $q;
+        this.$element = $element;
+        this.$attrs = $attrs;
+        this.dualListProvider = ouiDualListProvider;
+    }
+
+    $onInit () {
+        addDefaultParameter(this, "sourceListLabel", this.dualListProvider.translations.sourceListLabel);
+        addDefaultParameter(this, "targetListLabel", this.dualListProvider.translations.targetListLabel);
+        addDefaultParameter(this, "moveAllLabel", this.dualListProvider.translations.moveAllLabel);
+        addDefaultParameter(this, "removeAllLabel", this.dualListProvider.translations.removeAllLabel);
+        addDefaultParameter(this, "addLabel", this.dualListProvider.translations.addLabel);
+        addDefaultParameter(this, "sourceListEmptyLabel", this.dualListProvider.translations.sourceListEmptyLabel);
+        addDefaultParameter(this, "targetListEmptyLabel", this.dualListProvider.translations.targetListEmptyLabel);
+        addDefaultParameter(this, "property", null);
+        addBooleanParameter(this, "bulkActionEnabled");
+
+        this.sourceList = this.sourceList || [];
+        this.targetList = this.targetList || [];
+        this.onAdd = this.onAdd || null;
+        this.onRemove = this.onRemove || null;
+
+        this.loadingMap = {};
+        this.sourceListLoading = false;
+        this.targetListLoading = false;
+        this.loadSourceList();
+        this.loadTargetList();
+    }
+
+    $postLink () {
+        this.$element.addClass("oui-dual-list");
+        this.sourceCloseIcon = this.$element[0].querySelector(".oui-dual-list__source > .oui-dual-list__header > .oui-dual-list__header_toggle > .oui-dual-list__header_toggle_up");
+        this.sourceOpenIcon = this.$element[0].querySelector(".oui-dual-list__source > .oui-dual-list__header > .oui-dual-list__header_toggle > .oui-dual-list__header_toggle_down");
+        this.sourceContent = this.$element[0].querySelector(".oui-dual-list__source > .oui-dual-list__content");
+        this.targetCloseIcon = this.$element[0].querySelector(".oui-dual-list__target > .oui-dual-list__header > .oui-dual-list__header_toggle > .oui-dual-list__header_toggle_up");
+        this.targetOpenIcon = this.$element[0].querySelector(".oui-dual-list__target > .oui-dual-list__header > .oui-dual-list__header_toggle > .oui-dual-list__header_toggle_down");
+        this.targetContent = this.$element[0].querySelector(".oui-dual-list__target > .oui-dual-list__content");
+
+        angular.element(this.sourceCloseIcon).addClass("oui-dual-list__display-inline-block");
```

Comment:
We really should avoid using classnames with CSS property in its name.
Especially to toggle a display value that could be done in an AngularJS way. 

For example, here for your sourceCloseIcon, we could replace

```
<div class="oui-dual-list__source">
    <div class="oui-dual-list__header">
        [...]
        <span class="oui-dual-list__header_toggle">
            <i class="oui-icon oui-icon-chevron-up oui-dual-list__header_toggle_up"
                data-ng-click="$ctrl.onSourceContentClose()"
                aria-hidden="true"></i>
            <i class="oui-icon oui-icon-chevron-down oui-dual-list__header_toggle_down"
                data-ng-click="$ctrl.onSourceContentOpen()"
                aria-hidden="true"></i>
        </span>
    </div>
    <div class="oui-dual-list__content">
        [...]
    </div>
</div>
```
by
```
<div class="oui-dual-list__source">
    <div class="oui-dual-list__header">
        [...]
        <button class="oui-dual-list__header_toggle"
            ng-click="$ctrl.toggleSource()">
            <span class="oui-icon" ng-class="{
                    'oui-icon-chevron-up': $ctrl.isSourceOpen,
                    'oui-icon-chevron-down': !$ctrl.isSourceOpen
                }">
            </span>
        </button>
    </div>
    <div class="oui-dual-list__content" ng-show="$ctrl.isSourceOpen">
        [...]
    </div>
</div>

toggleSource () {
    this.isSourceOpen = !this.isSourceOpen;
}
```

Reply:
Good suggestion. I made required changes but I need to add a code in the component to detect device based on width. Hope it is fine.